opnode/rollup/driver/step.go

```diff
@@ -194,7 +176,7 @@ func (d *outputImpl) step(ctx context.Context, l2Head eth.BlockID, l2Finalized e
 			return last, fmt.Errorf("failed to extend L2 chain at block %d/%d of epoch %d: %w", i, len(batches), epoch, err)
 		}
 		last = payload.ID()
-		fc.HeadBlockHash = last.Hash
+		fc.HeadBlockHash = last.Hash // should be safe block, but geth is broken
```

Comment:
If this is the new tip of the chain, and it is derived from L1, then it should be setting **both** `fc.HeadBlockHash` and `fc.SafeBlockHash`. Maybe that works as expected with geth?

Reply:
Yes, it should be setting both (when working as a verifier). The thing that geth needed was setting head block.